metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -783,7 +788,8 @@ void handleNodeDeactivated(int brokerId, List<ApiMessageAndVersion> records) {
                 record.setIsr(Replicas.toList(newIsr));
                 if (partition.leader == brokerId) {
                     // The fenced node will no longer be the leader.
-                    int newLeader = bestLeader(partition.replicas, newIsr, false);
+                    int newLeader = bestLeader(partition.replicas, newIsr,
+                        strategizer.shouldBeUnclean(topic.name));
```

Comment:
If there is an unclean leader election, we need to reset the ISR to just the leader. Since this needs to be done in multiple places, perhaps we could change bestLeader() to return both the new leader and the isr.